src/art.rs

```diff
@@ -1211,10 +1424,22 @@ impl<P: KeyTrait, V: Clone> Tree<P, V> {
                     return (Some(Arc::new(new_node)), true);
                 }
                 (None, true) => {
-                    // The key has been found, so remove the corresponding
-                    // child link from the current node.
-                    let new_node = cur_node.delete_child(k);
-                    return (Some(Arc::new(new_node)), true);
+                    // The key has been found, so either remove the corresponding
+                    // child link from the current node if that's not its
+                    // last child, or pull up its inner Twig node into this
+                    // position in the tree, or simply return (None, true)
+                    // if there's nothing to occupy this position.
+                    if cur_node.num_children() == 1 {
```

Comment:
In this scenario

```
           Node4("key")          
          /            \         
    InnerTwig("2")     Node4("2")
                      |
                 Twig("abc")      
```
We are deleting the inner_twig from what I understand here, but what happens to the node4(2) in this scenario?

Reply:
Oh wait, the above scenario won't exist at this clause. Could you help explain this scenario a bit?